paper.md

```diff
@@ -5,128 +5,148 @@ tags:
-for the R community, by having extensibility as a core feature of the language.
-
-Natively, R provides tools to compile and embed Fortran, C, and C++ code. The
-binding happens through R's C-API. The official R-package repository CRAN[^CRAN] hosts
-several community-driven
-efforts for providing bindings to R, from many languages like Julia, Python, Java,
-JavaScript, etc.
+The programming language [Rust](https://www.rust-lang.org) continues to gain popularity with
+developers due to a strong emphasis on safety, performance and productivity [@perkelWhyScientistsAre2020].
+As a general-purpose, low-level programming language, Rust has a wide variety of potential uses
+in both commercial and research applications where performance is important. Commercial examples
+include web development and game development, and in the research domain Rust is increasingly being
+used in a wide range of contexts including change point detection [@JMLR:v24:22-0512], high-performance
+GIF encoding [@gifski_cran], and agent-based models of disease spread [@eval_rust_for_custom_abm; @epirust_paper; @forth_african_2022-1]. 
+<!-- Other references/examples of Rust usage in research would be useful! -->
+
+However, typical workflows in research domains including disease modelling are typically based on a higher-level
+programming language, due to lower barriers to usage and therefore wider adoption within these
+communities compared to low-level languages such as C++ and Rust. The statistical programming language 
+[R](https://www.r-project.org) is one of the most widely used
+high-level languages in research. R's official interpreter is written in C, and provides a C-API as well as 
+tools for building dynamic libraries in Fortran/C/C++ natively.
+<!-- On Windows, the R-project provides
+a toolchain, Rtools, which bundles a developer environment for Fortran, C and C++. -->
+<!-- The R-project provides documentation for developing extensions in [R-internals](https://cran.r-project.org/doc/manuals/R-ints.html)
+and [R-extenstions](https://cran.r-project.org/doc/manuals/R-exts.html). -->
+The Extending R book [@chambers2017extending] also describes interfacing with other languages
+such as Python and Julia. 
+
+A strength of R is the ecosystem of packages hosted at CRAN[^CRAN; https://cran.r-project.org], which are primarily written by research scientists,
+specialists and professionals. An important use case is as a front-end to working with other
+languages, for which automated tooling providing scaffolding / boilerplate code for integrating external languages
+are widely used. For example, embedding C++ code is a good way to resolve performance bottlenecks
+within R packages, and can be easily accomplished using cpp11 [@cpp11] and Rcpp [@rcpp_jss]. 
+Rust has similar performance to C++ but also has other beneficial features such as declarative memory management,
+providing compile-time guarantees for memory safety without the need for a garbage collector. 
```

Comment:
missing the word and before the last fragment

Reply:
Actually I intended that to be read as the compile-time guarantees for memory safety are provided by declarative memory management, not that the two are separate. Maybe adding 'which' would make it more obvious? ie

... declarative memory management, which provides compile-time guarantees ...

??

Reply:
Ah! Yeah, that makes sense!